tests/api/merge_node_test.cpp

```diff
@@ -0,0 +1,132 @@
+/*******************************************************************************
+ * tests/api/merge_node_test.cpp
+ *
+ * Part of Project thrill.
+ *
+ * Copyright (C) 2015 Timo Bingmann <tb@panthema.net>
+ * Copyright (C) 2015 Emanuel Jöbstl <emanuel.joebstl@gemail.com>
+ *
+ * This file has no license. Only Chunk Norris can compile it.
+ ******************************************************************************/
+
+#include <thrill/api/allgather.hpp>
+#include <thrill/api/generate.hpp>
+#include <thrill/api/size.hpp>
+#include <thrill/api/merge.hpp>
+#include <thrill/common/string.hpp>
+#include <gtest/gtest.h>
+#include <thrill/data/block_queue.hpp>
+
+#include <algorithm>
+#include <random>
+#include <string>
+#include <vector>
+
+using namespace thrill;
+
+using thrill::api::Context;
+using thrill::api::DIA;
+
+template <typename stackA, typename stackB>
+void DoMergeAndCheckResult(api::DIA<size_t, stackA> merge_input1, api::DIA<size_t, stackB> merge_input2, size_t expected_size, int num_workers) {
+        // merge
+        auto merge_result = merge_input1.Merge(
+            merge_input2, std::less<size_t>());
+
+        // check if order was kept while merging. 
+        int count = 0;
+        auto res = merge_result.Map([&count] (size_t in) { 
+                count++; 
+                return in; 
+        }).AllGather();
+
+        ASSERT_EQ(expected_size, res.size());
+
+        for (size_t i = 0; i != res.size() - 1; ++i) {
+            ASSERT_TRUE(res[i] <= res[i + 1]);
+        }
+
+        // REVIEW(ej): check CONTENTS of res as well!
```

Comment:
I think this REVIEW item is still very important and open.
Just calculate the expected output in the caller, and compare the res vector.